packages/flutter/ios/sentry_flutter/Sources/sentry_flutter_objc/SentryFlutterReplayScreenshotProvider.m

```diff
@@ -28,74 +27,73 @@ - (void)imageWithView:(UIView *_Nonnull)view
   // On iOS, we only have access to scope's replay ID, so we cannot detect buffer mode
   // If replay ID exists, it's always in active session mode (not buffering)
   BOOL replayIsBuffering = NO;
-  [self->channel
-      invokeMethod:@"captureReplayScreenshot"
-         arguments:@{
-           @"replayId" : replayId ? replayId : [NSNull null],
-           @"replayIsBuffering" : @(replayIsBuffering)
-         }
-            result:^(id value) {
-              if (value == nil) {
-                NSLog(@"SentryFlutterReplayScreenshotProvider received null "
-                      @"result. "
-                      @"Cannot capture a replay screenshot.");
-              } else if ([value isKindOfClass:[NSDictionary class]]) {
-                NSDictionary *dict = (NSDictionary *)value;
-                long address = ((NSNumber *)dict[@"address"]).longValue;
-                NSNumber *length = ((NSNumber *)dict[@"length"]);
-                NSNumber *width = ((NSNumber *)dict[@"width"]);
-                NSNumber *height = ((NSNumber *)dict[@"height"]);
-                NSData *data =
-                    [NSData dataWithBytesNoCopy:(void *)address
-                                         length:length.unsignedLongValue
-                                   freeWhenDone:TRUE];
 
-                // We expect rawRGBA, see docs for ImageByteFormat:
-                // https://api.flutter.dev/flutter/dart-ui/ImageByteFormat.html
-                // Unencoded bytes, in RGBA row-primary form with premultiplied
-                // alpha, 8 bits per channel.
-                static const int kBitsPerChannel = 8;
-                static const int kBytesPerPixel = 4;
-                assert(length.unsignedLongValue % kBytesPerPixel == 0);
+  if (self->callback == nil) {
+    NSLog(@"SentryFlutterReplayRecorderFFI has no callback. "
+          @"Cannot capture a replay screenshot.");
+    return;
+  }
+
+  self->callback(replayId, replayIsBuffering, ^(id value) {
+    if (value == nil) {
+      NSLog(@"SentryFlutterReplayRecorderFFI received null result. "
+            @"Cannot capture a replay screenshot.");
+    } else if ([value isKindOfClass:[NSDictionary class]]) {
```

Comment:
We can have two params with explicit types of `NSDictionary *` and `FlutterError`. But see this as optional if you just want to port this 1:1 for now.